### ISSUE TITLE
Automation Traces - Trace shifts and not inline

### ISSUE THREAD

**Issue by Blackshome** [home-assistant/frontend#19304]
### The problem

Best to just show you.

In HA

![3](https://github.com/home-assistant/core/assets/98072753/0696f740-7059-4660-b353-25db96947188)

The trace line is not inline as shown below.

![2](https://github.com/home-assistant/core/assets/98072753/683e6326-2bb9-4acd-8a43-e1b3c8a03ab1)

If we roll back to last release 2023.12.4 the trace line is inline as shown below.

![4](https://github.com/home-assistant/core/assets/98072753/71e00c62-e286-4dc6-b98e-6be3f3803584)

Hope this helps

Blacky :)

### What version of Home Assistant Core has the issue?

2024.1.1

### What was the last working version of Home Assistant Core?

2023.12.4

### What type of installation are you running?

Home Assistant OS

### Integration causing the issue

Automation

**Comment by frenck** [comment]
Can you provide the trace?

**Comment by karwosts** [comment]
Maybe related #19036, was a trace change made recently.

**Comment by karwosts** [comment]
Hmm yeah that does look maybe problematic. I made some simple test automation:

```
alias: trace test
description: ""
trigger: []
condition: []
action:
  - repeat:
      sequence:
        - delay:
            hours: 0
            minutes: 0
            seconds: 0
            milliseconds: 10
        - if:
            - condition: template
              value_template: "{{ repeat.index == 10 }}"
          then:
            - stop: ""
      while:
        - condition: template
          value_template: "{{ true }}"
    enabled: true
mode: single
```

Just run it, and the trace looks a little broken:
![image](https://github.com/home-assistant/frontend/assets/32912880/acd85543-974c-4553-b0bb-d11fd7ddb5cd)

If I put back the code that was removed in that PR, it looks better:
![image](https://github.com/home-assistant/frontend/assets/32912880/a72abaa6-e189-47da-9129-0336a77cc307)

**Comment by Blackshome** [comment]
@frenck I have rolled back to 2023.12.4 waiting for this fix as I use traces all the time. The trace is gone. 

I do use Firefox browser and the automation was one of my [blueprints](https://community.home-assistant.io/t/sensor-light-motion-sensor-door-sensor-sun-elevation-lux-value-scenes-time-light-control-device-tracker-night-lights/481048/1). It was in dynamic light control.

Looks like @karwosts is onto it though.

Blacky :)

**Comment by Blackshome** [comment]
@frenck @karwosts Any news on this yet?

**Comment by karwosts** [comment]
I think this will be resolved by #19557, but let me know after the next release if its not looking better.

**Comment by Blackshome** [comment]
Can confirm this is all fixed in this months release with some nice icons added... nice work, thanks for your help.
